grid-client/subi/substrate_manager.go

```diff
@@ -79,6 +87,67 @@ type SubstrateImpl struct {
 	m sync.Mutex
 }
 
+var _ SubstrateExt = (*SubstrateImpl)(nil)
+
+// NewIdentityFromSr25519Phrase returns the identity from mnemonic
+func (s *SubstrateImpl) NewIdentityFromSr25519Phrase(mnemonic string) (substrate.Identity, error) {
+	return substrate.NewIdentityFromSr25519Phrase(mnemonic)
+}
+
+// AcceptTermsAndConditions accepts terms and conditions
+func (s *SubstrateImpl) AcceptTermsAndConditions(identity substrate.Identity, docLink string, docHash string) error {
+	return s.Substrate.AcceptTermsAndConditions(identity, docLink, docHash)
+}
+
+// CreateTwin creates a twin and returns its twin ID
+func (s *SubstrateImpl) CreateTwin(identity substrate.Identity, relay string, pk []byte) (uint32, error) {
+	return s.Substrate.CreateTwin(identity, relay, pk)
+}
+
+// CreateRentContract creates a rent contract
+func (s *SubstrateImpl) CreateRentContract(mnemonic string, nodeID uint32, solutionProviderID *uint64) (uint64, error) {
+	identity, err := s.NewIdentityFromSr25519Phrase(mnemonic)
+	if err != nil {
+		return 0, err
+	}
+
+	return s.Substrate.CreateRentContract(identity, nodeID, solutionProviderID)
+}
+
+// TransferTFTsFromSystem transfer balance to users' account
+func (s *SubstrateImpl) TransferTFTsFromSystem(tftBalance uint64, userMnemonic string, systemMnemonic string) error {
+	// Create identity of user from mnemonic
+	userIdentity, err := substrate.NewIdentityFromSr25519Phrase(userMnemonic)
```

Comment:
You should use the wrapper method `s.NewIdentityFromSr25519Phrase` instead of calling the lower-level implementation directly.
If there’s a reason or a special case where direct use is justified, please document it explicitly.